app/instrumentation-tests/src/org/commcare/androidTests/AdvancedSettingsTests.kt

```diff
@@ -0,0 +1,212 @@
+package org.commcare.androidTests
+
+
+import android.content.Intent
+import android.util.Log
+import android.widget.TextView
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.typeText
+import androidx.test.espresso.assertion.PositionAssertions.isAbove
+import androidx.test.espresso.assertion.PositionAssertions.isBelow
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.runner.intent.IntentCallback
+import androidx.test.runner.intent.IntentMonitorRegistry
+import androidx.test.uiautomator.UiDevice
+import junit.framework.TestCase
+import org.commcare.CommCareApplication
+import org.commcare.annotations.BrowserstackTests
+import org.commcare.dalvik.R
+import org.commcare.utils.CustomMatchers
+import org.commcare.utils.InstrumentationUtility
+import org.commcare.utils.isDisplayed
+import org.commcare.utils.isPresent
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.*
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import kotlin.random.Random
+
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+@BrowserstackTests
+class AdvancedSettingsTests : BaseTest() {
+
+    companion object {
+        const val CCZ_NAME = "advanced_settings.ccz"
+        const val APP_NAME = "Advanced Settings"
+        val NAME = "test"+(LocalDateTime.now()).format(DateTimeFormatter.BASIC_ISO_DATE)+List(1){ Random.nextInt(1000, 9999) }
+    }
+
+    @Before
+    fun setup() {
+
+        if (CommCareApplication.instance().currentApp == null) {
+            InstrumentationUtility.installApp(CCZ_NAME)
+        } else {
+            InstrumentationUtility.uninstallCurrentApp()
+            InstrumentationUtility.installApp(CCZ_NAME)
+            Espresso.pressBack()
+        }
+        InstrumentationUtility.login("test1", "123")
+    }
+
+    @After
+    fun tearDown() {
+        InstrumentationUtility.logout()
+    }
+
+    @Test
+    fun testAdvancedSettings() {
```

Comment:
can we oragnise this test into different methods and have each one of them test one of the custom properties from the test plan. Right now it's quite confusing what all this test is trying to do.